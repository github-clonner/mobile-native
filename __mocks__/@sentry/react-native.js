export const captureMessage = jest.fn();
export default {};
