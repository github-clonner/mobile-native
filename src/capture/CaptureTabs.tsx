//@ts-nocheck
import React, { Component } from 'react';

import {
  StyleSheet,
  View,
  Text,
  TouchableHighlight,
  Platform,
} from 'react-native';

import Icon from 'react-native-vector-icons/Ionicons';

import { CommonStyle } from '../styles/Common';
import colors from '../styles/Colors';
import attachmentService from '../common/services/attachment.service';
import i18n from '../common/services/i18n.service';
import ThemedStyles from '../styles/ThemedStyles';

/**
 * Capture tab
 */
export default class CaptureTab extends Component {
  state = {
    screen: 'gallery',
    imageUri: '',
    attachmentGuid: '',
    attachmentDone: false,
  };

  static navigationOptions = {
    header: null,
    tabBarIcon: ({ tintColor }) => (
      <Icon name="md-radio-button-on" size={24} color={tintColor} />
    ),
  };

  /**
   * Render
   */
  render() {
    return (
      <View style={CommonStyle.flexContainer}>
        {this.state.screen != 'poster' ? (
          <View style={{ height: 125 }}>
            <View
              style={{
                flex: 1,
                flexDirection: 'row',
                paddingLeft: 1,
                paddingRight: 1,
              }}>
              <TouchableHighlight onPress={this.gallery} style={styles.buttons}>
                <View style={styles.buttonsWrapper}>
                  <Icon
                    name="md-photos"
                    size={36}
                    style={ThemedStyles.style.colorIcon}
                  />
                  <Text style={[styles.labels, ThemedStyles.style.colorIcon]}>
                    {i18n.t('capture.gallery')}
                  </Text>
                </View>
              </TouchableHighlight>
              <TouchableHighlight onPress={this.photo} style={styles.buttons}>
                <View style={styles.buttonsWrapper}>
                  <Icon
                    name="md-camera"
                    size={36}
                    style={ThemedStyles.style.colorIcon}
                  />
                  <Text style={[styles.labels, ThemedStyles.style.colorIcon]}>
                    {i18n.t('capture.photo')}
                  </Text>
                </View>
              </TouchableHighlight>
              <TouchableHighlight onPress={this.video} style={styles.buttons}>
                <View style={styles.buttonsWrapper}>
                  <Icon
                    name="md-videocam"
                    size={36}
                    style={ThemedStyles.style.colorIcon}
                  />
                  <Text style={[styles.labels, ThemedStyles.style.colorIcon]}>
                    {i18n.t('capture.video')}
                  </Text>
                </View>
              </TouchableHighlight>
            </View>
          </View>
        ) : (
          <View></View>
        )}
      </View>
    );
  }

  video = async () => {
    try {
      const response = await attachmentService.video();
      if (response) this.props.onSelectedMedia(response);
    } catch (e) {
      alert(e);
    }
  };

  photo = async () => {
    try {
      const response = await attachmentService.photo();
      if (response) this.props.onSelectedMedia(response);
    } catch (e) {
      alert(e);
    }
  };

  /**
   * Open gallery
   */
  gallery = async () => {
    try {
      const response = await attachmentService.gallery('mixed');
      if (response) this.props.onSelectedMedia(response);
    } catch (e) {
      alert(e);
    }
  };

  /**
   * Return tab style
   * @param {string} value
   */
  tabButtonStyle(value) {
    if (value === this.state.screen) {
      return {
        flex: 1,
        borderRadius: 0,
      };
    } else {
      return {
        flex: 1,
        borderRadius: 0,
      };
    }
  }

  /**
   * Reset state to default
   */
  resetState() {
    this.setState({
      screen: 'gallery',
      imageUri: '',
      attachmentGuid: '',
      attachmentDone: false,
    });
  }

  /**
   * Move to capture screen
   */
  moveToCaptureScreen() {
    this.setState({ screen: 'captureImage' });
  }
}

const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
  },
  labels: {
    letterSpacing: 1.25,
    color: '#444',
  },
  buttons: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    alignContent: 'center',
    margin: 1,
  },
  buttonsWrapper: {
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    alignContent: 'center',
  },
  selectedButton: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    borderBottomWidth: 3,
    backgroundColor: '#FFF',
    borderColor: colors.primary,
  },
});
