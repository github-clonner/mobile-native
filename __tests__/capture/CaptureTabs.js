import 'react-native';
import React from 'react';
import {
  TouchableHighlight,
  Platform
} from 'react-native';
import { shallow } from 'enzyme';
import CaptureTabs from '../../src/capture/CaptureTabs';
import attachmentService from '../../src/common/services/attachment.service';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const imageResponse = {
  uri: 'file://someimage.jpg',
  type: 'image/jpeg',
  fileName: 'image.jpg'
}

/**
 * Tests
 */
describe('cature tabs component', () => {

  it('should renders correctly', () => {
    const tabs = renderer.create(
      <CaptureTabs  />
    ).toJSON();
    expect(tabs).toMatchSnapshot();
  });

  it('should calls attachment.gallery and return the media', async(done) => {

    // mock the function
    attachmentService.gallery = jest.fn();
    attachmentService.gallery.mockResolvedValue(imageResponse);

    const mockFn = jest.fn();

    Platform.OS = 'ios';

    try {
      const wrapper = shallow(
        <CaptureTabs  onSelectedMedia={mockFn}/>
      );

      // find TouchableOpacity (rendered images in lists)
      const tabs = wrapper.find(TouchableHighlight);

      // simulate press on first tab (gallery)
      tabs.at(0).simulate('press');

      // we use set inmediate to run the check after the async function is called
      setImmediate(() => {
        // expect fn to be called once
        expect(mockFn).toBeCalled();
        done();
      }, 0);
    } catch(e) {
      done.fail(e);
    }
  });

  it('should calls attachment.photo and return the media', async(done) => {

    // mock the function
    attachmentService.photo = jest.fn();
    attachmentService.photo.mockResolvedValue(imageResponse);

    const mockFn = jest.fn();

    try {
      const wrapper = shallow(
        <CaptureTabs  onSelectedMedia={mockFn}/>
      );

      // find TouchableOpacity (rendered images in lists)
      const tabs = wrapper.find(TouchableHighlight);

      // simulate press on second tab (gallery)
      tabs.at(1).simulate('press');

      // we use set inmediate to run the check after the async function is called
      setImmediate(() => {
        // expect fn to be called once
        expect(mockFn).toBeCalled();
        done();
      }, 0);
    } catch(e) {
      done.fail(e);
    }
  });

  it('should calls attachment.video and return the media', async(done) => {

    // mock the function
    attachmentService.video = jest.fn();
    attachmentService.video.mockResolvedValue(imageResponse);

    const mockFn = jest.fn();

    try {
      const wrapper = shallow(
        <CaptureTabs  onSelectedMedia={mockFn}/>
      );

      // find TouchableOpacity (rendered images in lists)
      const tabs = wrapper.find(TouchableHighlight);

      // simulate press on third tab (gallery)
      tabs.at(2).simulate('press');

      // we use set inmediate to run the check after the async function is called
      setImmediate(() => {
        // expect fn to be called once
        expect(mockFn).toBeCalled();
        done();
      }, 0);
    } catch(e) {
      done.fail(e);
    }
  });

});
