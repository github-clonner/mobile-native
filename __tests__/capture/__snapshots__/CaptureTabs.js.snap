// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cature tabs component should renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": 125,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "paddingLeft": 1,
          "paddingRight": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "margin": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 36,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#444",
                  "letterSpacing": 1.25,
                },
                undefined,
              ]
            }
          >
            Gallery
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "margin": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 36,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#444",
                  "letterSpacing": 1.25,
                },
                undefined,
              ]
            }
          >
            Photo
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "margin": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 36,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#444",
                  "letterSpacing": 1.25,
                },
                undefined,
              ]
            }
          >
            Video
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
